### PR TITLE
Updated makefile by adding more targets for developer.

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
NONE
```

/cc @hougangliu @gaocegege @richardsliu

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/575)

### CODE REVIEW

File: docs/developer-guide.md [hunk 1]
```diff
@@ -8,16 +8,22 @@
...
+
+```
+make update
 ```
 
 You can build all images from source.
 
 ```bash
-./scripts/v1alpha1/build.sh
+make build
```
Comment (hougangliu) [inline docs/developer-guide.md]: can you refactor it to add version arg to `bash scripts/v1alpha1/build.sh` or `bash scripts/v1alpha2/build.sh`

Comment (gyliu513) [reply docs/developer-guide.md]: My thinking is as follows, I will try to update `makefile` when `v1alpha2` become stable by making `make build` to build `v1alpha2` and introduce another `make build-v1alpha1` to build the old `v1alpha1` version.

But for now, I want to use `make build` default to `v1alpha1` as `v1alpha2` is not ready yet, comments?

Comment (hougangliu) [reply docs/developer-guide.md]: SGTM

**Pull request merged**